src/cs/Ssh/Progress.cs

```diff
@@ -24,4 +24,18 @@ public enum Progress
 	CompletedKeyExchange,
 	StartingSessionAuthentication,
 	CompletedSessionAuthentication,
+
+	// Port forwarding
+	StartingRefreshPorts,
+	CompletedRefreshPorts,
+
+	// Tunnel service requests
```

Comment:
Please avoid adding anything specific to the tunnel service in this repo. I try to keep this library generally useful for any applications working with SSH. While these constants won't cause any problems, I'd rather not set a precedent.

Reply:
Sounds good. I'll likely abandon these changes and have a completely different TunnelProgress enumerartion for the TunnelService specific progress events.